examples/cpakdemo/cpakdemo.c

```diff
@@ -0,0 +1,620 @@
+    sprintf(path, "%s%s", prefix, fname);
+
+    printf("Opening file %s\n", fname);
+    FILE *f = fopen(path, "wb");
+    if (f == NULL) {
+        ok = false;
+        printf("Unable to open file\n");
+        printf("(%d) %s\n", errno, strerror(errno));
+    } else {
+        const uint8_t content[] = {
+            'c', 'p', 'a', 'k', '_', 'd', 'e', 'm', 'o',
+        };
+        size_t size = sizeof(content);
+
+        printf("Writing content\n");
+        int nwritten = fwrite(content, 1, size, f);
+        if (nwritten != size) {
+            ok = false;
+            printf("Unable to write requested amount of bytes: %u != %u\n", nwritten, size);
+            printf("(%d) %s\n", errno, strerror(errno));
+        }
+
+        err = fclose(f);
+        if (err < 0) {
+            ok = false;
+            printf("Unable to close file (%d)\n", err);
+            printf("(%d) %s\n", errno, strerror(errno));
+        }
+    }
+
+    err = cpakfs_unmount(port);
+    if (err < 0) {
+        ok = false;
+        printf("Unable to unmount filesystem (%d)\n", err);
+        printf("(%d) %s\n", errno, strerror(errno));
+    }
+
+    if (ok) {
+        printf("File %s created\n", fname);
+    }
+
+    print_actions();
+    console_render();
+    return STATE_WAIT_ACTION;
+}
+
+
+enum program_state_type state_delete_file(joypad_port_t port, const char* prefix, const char* fname) {
+    console_clear();
+    print_working_controller_message(port);
+    printf("Deleting file\n");
+
+    bool ok = true;
+
+    int err = cpakfs_mount(port, prefix);
+    if (err < 0) {
+        ok = false;
+        printf("Unable to mount cpak filesystem (%d)\n", err);
+        printf("(%d) %s\n", errno, strerror(errno));
+        print_actions();
+        console_render();
+        return STATE_WAIT_ACTION;
+    }
+
+    char path[255 + 8];
+    sprintf(path, "%s%s", prefix, fname);
+
+    err = remove(path);
+    if (err != 0) {
+        ok = false;
+        printf("Unable to remove file (%d)\n", err);
+        printf("(%d) %s\n", errno, strerror(errno));
+    }
+
+    err = cpakfs_unmount(port);
+    if (err < 0) {
+        ok = false;
+        printf("Unable to unmount filesystem (%d)\n", err);
+        printf("(%d) %s\n", errno, strerror(errno));
+    }
+
+    if (ok) {
+        printf("File %s deleted\n", fname);
+    }
+
+    print_actions();
+    console_render();
+    return STATE_WAIT_ACTION;
+}
+
+
+int main(void)
+{
+    // Only handle controller pak from player 1
+    const joypad_port_t port = JOYPAD_PORT_1;
+
+    // Recommended prefix is 'cpakN:/' with N being the player number (1..4)
+    char prefix[8];
+    sprintf(prefix, "cpak%d:/", port + 1);
+
+    // Full name must respect the GAME.PU-filename.ext format
+    // To make gamecode match the one from ROM Header (NED\x00)
+    // we use the hex encoding of gamecode because NED\x00
+    // contains a \x00 which is not useable in C strings.
+    const char* fname = "4e454400.XX-TEST.A";
```

Comment:
I've changed the fname to make gamecode match the one from the ROM header.
Given that it contains a \x00 character, I've resorted to use the hex encoding for gamecode.
I've tried "NED\x01.XX-TEST.A" but that didn't seem to work (I thought it would).

I think it is somewhat important to showcase proper gamecode usage for homebrews which will almost all use NED\x00 game code. What do you think ?